caiman/source_extraction/cnmf/spatial.py

```diff
@@ -1050,22 +1050,24 @@ def computing_indicator(Y, A_in, b, C, f, nb, method, dims, min_size, max_size,
             px = (np.sum(dist_indicator, axis=1) > 0)
             not_px = ~px
 
-            if nb>1:
-                    f = NMF(nb, init='nndsvda').fit(np.maximum(Y[not_px, :], 0)).components_
-            else:
-                if Y.shape[-1] < 30000:
-                    f = Y[not_px, :].mean(0)
+            if nb > 0:
+                if nb > 1:
+                        f = NMF(nb, init='nndsvda').fit(np.maximum(Y[not_px, :], 0)).components_
                 else:
-                    print('estimating f')
-                    f = 0
-                    for xxx in np.where(not_px)[0]:
-                        f += Y[xxx]
-                    f /= not_px.sum()
-
-            f = np.atleast_2d(f)
-
-            Y_resf = np.dot(Y, f.T)
-            b = np.maximum(Y_resf, 0) / (np.linalg.norm(f)**2)
+                    if Y.shape[-1] < 30000:
```

Comment:
We may want to consider making this threshold overridable or at least less magical